### PR TITLE
Add strings for NavigationBar on relevant WebUI data sources directly

### PR DESCRIPTION
**Pull request by petemill**
This fixes an intermittent build error due to the generated_resources being included in web_ui_util (inside the //ui/base project)

This fixes a layer-violation for dependencies where we were using a convenience function in the base //ui project in order to set the strings on all WebUI data sources. With this refactor, the dependencies are already in place from the chromium ui project (containing the webui data sources) to the brave webui project (containing the new navigation bar data provider).

This new NavigationBarDataProvider is not much of a class, in that it only contains a single static function, but it follows the chromium practice of splitting data in to functional classes, and leaving it up to the provider as to whether it adds strings, functions or observers to the WebUI.

Fix https://github.com/brave/brave-browser/issues/4343

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
Dependency error is happened.
```
12:50:32  ERROR at //build/config/BUILDCONFIG.gn:563:5: assert_no_deps failed.
12:50:32      target(_target_type, target_name) {
12:50:32      ^----------------------------------
12:50:32  //components:components_unittests has an assert_no_deps entry:
12:50:32    //chrome/*
12:50:32  which fails for the dependency path:
12:50:32    //components:components_unittests ->
12:50:32    //components/autofill/core/browser:unit_tests ->
12:50:32    //components/autofill/core/browser:browser ->
12:50:32    //components/autofill/core/common:common ->
12:50:32    //ui/base:base ->
12:50:32    //brave/ui/base:chromium_deps ->
12:50:32    //chrome/app:generated_resources
```

**Comment by bsclifton** [comment]
got the same error as above

**Comment by bsclifton** [comment]
Created https://github.com/brave/brave-browser/issues/4343 to track the problem

**Comment by petemill** [comment]
Updated with a completely different approach. Running tests now

File: chromium_src/ui/base/webui/web_ui_util.cc [hunk 1]
```diff
@@ -3,48 +3,14 @@
...
-#include "components/strings/grit/components_strings.h"
-#include "ui/base/l10n/l10n_util.h"
-
-namespace webui {
-
-void SetLoadTimeDataDefaults(const std::string& app_locale,
-                             base::DictionaryValue* localized_strings) {
-
-  SetLoadTimeDataDefaults_ChromiumImpl(app_locale, localized_strings);
 #if !defined(OS_ANDROID)
```
Comment (simonhong) [inline chromium_src/ui/base/webui/web_ui_util.cc]: This define pair can be removed

File: chromium_src/chrome/browser/ui/webui/history_ui.cc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace {
+
+void BraveCustomizeHistoryDataSource(content::WebUIDataSource* source) {
+  NavigationBarDataProvider::Initialize(source);
+}
+
+}  // namespace
+
+#include "../../../../../chrome/browser/ui/webui/history_ui.cc"
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/history_ui.cc]: lint?

File: chromium_src/chrome/browser/ui/webui/downloads/downloads_ui.cc [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace {
+
+void BraveCustomizeDownloadsDataSource(content::WebUIDataSource* source) {
+  NavigationBarDataProvider::Initialize(source);
+}
+
+}  // namespace
+
+#include "../../../../../../chrome/browser/ui/webui/downloads/downloads_ui.cc"  // NOLINT
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/downloads/downloads_ui.cc]: lint fail by no last empty line?

File: browser/ui/webui/navigation_bar_data_provider.cc [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+void NavigationBarDataProvider::Initialize(content::WebUIDataSource* source) {
+  source->AddLocalizedString(
+    "brToolbarSettingsTitle", IDS_SETTINGS_SETTINGS);
+  source->AddLocalizedString(
+    "brToolbarBookmarksTitle", IDS_BOOKMARK_MANAGER_TITLE);
+  source->AddLocalizedString(
+    "brToolbarDownloadsTitle", IDS_DOWNLOAD_TITLE);
+  source->AddLocalizedString(
+    "brToolbarHistoryTitle", IDS_HISTORY_TITLE);
+}
```
Comment (simonhong) [inline browser/ui/webui/navigation_bar_data_provider.cc]: lint?

File: chromium_src/chrome/browser/ui/webui/bookmarks/bookmarks_ui.cc [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace {
+
+void BraveCustomizeBookmarksDataSource(content::WebUIDataSource* source) {
+  NavigationBarDataProvider::Initialize(source);
+}
+
+}  // namespace
+
+#include "../../../../../../chrome/browser/ui/webui/bookmarks/bookmarks_ui.cc"
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/bookmarks/bookmarks_ui.cc]: lint

**Pull request merged**